protocol/src/main/resources/protocol.xml

```diff
@@ -132,6 +136,15 @@
   <sbe:message name="AdminResponse" id="41">
   </sbe:message>
 
+  <sbe:message name="BackupRequest" id="42">
+    <field name="partitionId" id="1" type="uint16"/>
+    <field name="type" id="2" type="BackupRequestType"/>
+    <field name="checkpointId" id = "3" type="int64"/>
```

Comment:
:wrench: I think this should be `uint64`, -1 is not allowed here, right?

:question: We are calling this a BackupRequest, should we then also call this `backupId`? Broker knows that to take a backup with `backupId`, it must write a checkpoint create command with that same id. The gateway doesn't need to know this.

Reply:
:+1: Will rename it to `backupId`.

In the broker -1 is a valid value to represent non-existing checkpoint. So I think, even if the user cannot create a checkpoint with -1, it should be int64. Otherwise if the user provides a uint64 and the broker interprets using int64 it wouldn't be correct, right?